x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/CcrStatsAction.java

```diff
@@ -0,0 +1,179 @@
+                final List<TaskResponse> taskResponses) {
+            super(taskFailures, nodeFailures);
+            this.taskResponses = taskResponses;
+        }
+
+        @Override
+        public XContentBuilder toXContent(final XContentBuilder builder, final Params params) throws IOException {
+            // sort by index name, then shard ID
+            final Map<String, Map<Integer, TaskResponse>> taskResponsesByIndex = new TreeMap<>();
+            for (final TaskResponse taskResponse : taskResponses) {
+                taskResponsesByIndex.computeIfAbsent(
+                        taskResponse.followerShardId().getIndexName(),
+                        k -> new TreeMap<>()).put(taskResponse.followerShardId().getId(), taskResponse);
+            }
+            builder.startObject();
+            {
+                for (final Map.Entry<String, Map<Integer, TaskResponse>> index : taskResponsesByIndex.entrySet()) {
+                    builder.startArray(index.getKey());
+                    {
+                        for (final Map.Entry<Integer, TaskResponse> shard : index.getValue().entrySet()) {
+                            shard.getValue().status().toXContent(builder, params);
+                        }
+                    }
+                    builder.endArray();
+                }
+            }
+            builder.endObject();
+            return builder;
+        }
+    }
+
+    public static class TasksRequest extends BaseTasksRequest<TasksRequest> implements IndicesRequest {
+
+        private String[] indices;
+
+        @Override
+        public String[] indices() {
+            return indices;
+        }
+
+        public void setIndices(final String[] indices) {
+            this.indices = indices;
+        }
+
+        private IndicesOptions indicesOptions = IndicesOptions.strictExpandOpenAndForbidClosed();
+
+        @Override
+        public IndicesOptions indicesOptions() {
+            return indicesOptions;
+        }
+
+        public void setIndicesOptions(final IndicesOptions indicesOptions) {
+            this.indicesOptions = indicesOptions;
+        }
+
+        @Override
+        public boolean match(final Task task) {
+            /*
+             * This is a limitation of the current tasks API. When the transport action is executed, the tasks API invokes this match method
+             * to find the tasks on which to execute the task-level operation (see TransportTasksAction#nodeOperation and
+             * TransportTasksAction#taskOperation). If we do the matching here, then we can not match index patterns. Therefore, we defer
+             * deciding whether or not the task matches the request to the transport action (see TransportCcrStatsAction#taskOperation)
+             * where we can decide on the basis of the cluster state whether or not the task matches any of the index patterns in the
+             * request.
+             */
+            return task instanceof ShardFollowNodeTask;
```

Comment:
I believe this can be an assertion now? we filter for the task instance type in the transport action.

Reply:
I would support throwing `UnsupportedOperationException`.

Reply:
I pushed b9c5429fe8330b32add40c0c33cd4c764eecc018.